File: python/680.py
class Solution:
    def validPalindrome(self, s: str) -> bool:
        def check(low,high):
            i,j=low,high
            while i<j:
                if s[i]!=s[j]:
                    return False;
                i+=1
                j-=1
            return True
        low,high=0,len(s)-1
        while low<high:
            if s[low]==s[high]:
                low+=1
                high-=1
            else:
                return check(low,high-1) or check(low+1,high)
        return True



s=Solution()
s.validPalindrome("abc")

File: cpp/680.cpp
#include "leetcode.h"
class Solution {
public:
    bool check(string &s,int low,int high)
    {
        for(int i=low,j=high;i<j;i++,j--)
        {
            if (s[i]!=s[j])
            {
                return false;
            }
            
        }
        return true;
    }
    bool validPalindrome(string s) {
        int size=s.size();
        for(int i=0,j=size-1;i<j;i++,j--)
        {
            if(s[i]==s[j])
            {
                continue;
            }
            else
            {
                return check(s,i,j-1) || check(s,i+1,j);
            }
            
        }
        return true;
    }
};

int main()
{
    Solution s;
    int irtn=0;
    return irtn=s.validPalindrome("abca");
}